src/status_im/utils/pre_receiver.cljs

```diff
@@ -0,0 +1,44 @@
+(ns status-im.utils.pre-receiver
+  (:require-macros [cljs.core.async.macros :as async])
+  (:require [cljs.core.async :as async]))
+
+;; See status-im.test.utils.pre-receiver for justification.
+
+;; TODO(oskarth): Doesn't belong here, mocking real add-msg - fix signature
+(defn- add-message [{:keys [id clock-value] :as msg}]
+  (println "add-message:" id clock-value))
+
+(defn- delay-message [msg out ms]
+  (async/go (async/<! (async/timeout ms))
+            (async/put! out msg)))
+
+(defn- earliest-clock-value-seen? [seen id clock-value]
+  (->> seen
+       (filter (fn [[_ x]] (= x id)))
+       sort
+       ffirst
+       (= clock-value)))
+
+;; TODO: Update docstring after add-message fixed
+(defn start!
```

Comment:
Should we write a tech debt bounty extracting the rolling window async mechanism into async utils namespace ?

Reply:
@yenda We could, I just don't see exactly how it'd be different than this namespace (without earliest-clock-value-seen? fn). What is the tech debt in your opinion?

Reply:
The delay-msg fn looks pretty canonical to me, and the pre-receiver could maybe be more generalized, but it also seems quite tight with specific problem. Do we have another use case than this in mind right now?